### ISSUE TITLE
No access to the profile after the PIN change attempt

### ISSUE THREAD

**Issue by drehzahl** [iotaledger/firefly#1491]
### Issue description

The PIN change has reported that an error has occurred.
Since then, I can't access the Firefly app with either the old or the new PIN.

### Firefly version

Firefly v1.2.0

### OS details

Windows Home 21H1 10.0.19043

### Can the issue be reliably reproduced?

Yes

### Steps to reproduce the issue

1. Enter PIN to login
2. enter Firefly Options
3. Security PIN Change
4. Enter old PIN and double new PIN hit Enter
5. then an error has occurred

### Expected behaviour

New pin should bet set.
Login still possible.

### Actual behaviour

1. Nothing is changed. Error message is visible.
2. after logging out. The profile cannot accessed anymore.

### Errors

Thu, 23 Sep 2021 18:03:32 GMT
RecordDecrypt: failed to decrypt record: CipherError { alg: "XCHACHA20-POLY1305" }

Wed, 22 Sep 2021 13:03:41 GMT
RecordDecrypt: failed to decrypt record: CipherError { alg: "XCHACHA20-POLY1305" }

Wed, 22 Sep 2021 09:15:19 GMT
RecordDecrypt: failed to decrypt record: CipherError { alg: "XCHACHA20-POLY1305" }

Wed, 22 Sep 2021 09:03:23 GMT
RecordDecrypt: failed to decrypt record: CipherError { alg: "XCHACHA20-POLY1305" }

Wed, 22 Sep 2021 07:25:37 GMT
RecordDecrypt: failed to decrypt record: CipherError { alg: "XCHACHA20-POLY1305" }

Wed, 22 Sep 2021 07:21:57 GMT
RecordDecrypt: failed to decrypt record: CipherError { alg: "XCHACHA20-POLY1305" }

Wed, 22 Sep 2021 07:09:23 GMT
RecordDecrypt: failed to decrypt record: CipherError { alg: "XCHACHA20-POLY1305" }

Wed, 22 Sep 2021 07:07:18 GMT
RecordDecrypt: failed to decrypt record: CipherError { alg: "XCHACHA20-POLY1305" }

Wed, 22 Sep 2021 06:50:34 GMT
RecordDecrypt: failed to decrypt record: CipherError { alg: "XCHACHA20-POLY1305" }

Wed, 22 Sep 2021 06:50:00 GMT
RecordDecrypt: failed to decrypt record: CipherError { alg: "XCHACHA20-POLY1305" }

Wed, 22 Sep 2021 06:48:48 GMT
RecordDecrypt: failed to decrypt record: CipherError { alg: "XCHACHA20-POLY1305" }

Wed, 22 Sep 2021 06:45:46 GMT
RecordDecrypt: failed to decrypt record: CipherError { alg: "XCHACHA20-POLY1305" }

Wed, 22 Sep 2021 06:42:33 GMT
RecordDecrypt: failed to decrypt record: CipherError { alg: "XCHACHA20-POLY1305" }

Wed, 22 Sep 2021 06:41:57 GMT
RecordDecrypt: failed to decrypt record: CipherError { alg: "XCHACHA20-POLY1305" }

Wed, 22 Sep 2021 06:41:38 GMT
RecordDecrypt: failed to decrypt record: CipherError { alg: "XCHACHA20-POLY1305" }

**Comment by maxwellmattryan** [comment]
Hey @drehzahl, nice find. Are there more specific repro steps that you did while changing the PIN?

**Comment by drehzahl** [comment]
> Hey @drehzahl, nice find. Are there more specific repro steps that you did while changing the PIN?

I have updated it

**Comment by Yanev1** [comment]
I have a similar issue, but I've never changed my PIN. It just stopped accepting it. Is it testing old features when fixing other bugs asking too much??

**Comment by Spidyisback** [comment]
Yes, I also think this is a "NO GO" for a new wallet. Had the same problem, even though I didn't change the pincode (and I remembered it). It just said "Something went wrong" and in the error log pretty much what Diego-8419 posted. The way I recovered my funds is by just creating a new profile and migrating again from Ledger backup. That worked pretty fast and smooth and it seems like it's now accepting my pin. At least for now ;-)
